wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/jpa/WorkspaceTckModule.java

```diff
@@ -11,25 +11,74 @@
 package org.eclipse.che.api.workspace.server.jpa;
 
 import com.google.inject.TypeLiteral;
-import com.google.inject.persist.PersistService;
+import com.google.inject.persist.Transactional;
 import com.google.inject.persist.jpa.JpaPersistModule;
 
 import org.eclipse.che.api.core.jdbc.jpa.guice.JpaInitializer;
+import org.eclipse.che.api.machine.server.model.impl.AclEntryImpl;
+import org.eclipse.che.api.user.server.model.impl.UserImpl;
 import org.eclipse.che.api.workspace.server.model.impl.WorkspaceImpl;
+import org.eclipse.che.api.workspace.server.model.impl.stack.StackImpl;
+import org.eclipse.che.api.workspace.server.spi.StackDao;
 import org.eclipse.che.api.workspace.server.spi.WorkspaceDao;
 import org.eclipse.che.commons.test.tck.TckModule;
+import org.eclipse.che.commons.test.tck.repository.JpaTckRepository;
 import org.eclipse.che.commons.test.tck.repository.TckRepository;
+import org.eclipse.che.commons.test.tck.repository.TckRepositoryException;
 
 import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.persistence.EntityManager;
+import java.util.Collection;
 
+/**
+ * @author Yevhenii Voevodin
+ */
 public class WorkspaceTckModule extends TckModule {
 
     @Override
     protected void configure() {
-        bind(WorkspaceDao.class).to(JpaWorkspaceDao.class);
-        bind(new TypeLiteral<TckRepository<WorkspaceImpl>>(){}).to(WorkspaceTckRepository.class);
         install(new JpaPersistModule("main"));
         bind(JpaInitializer.class).asEagerSingleton();
         bind(org.eclipse.che.api.core.h2.jdbc.jpa.eclipselink.H2ExceptionHandler.class);
+
+        bind(new TypeLiteral<TckRepository<WorkspaceImpl>>() {}).to(WorkspaceTckRepository.class);
+        bind(new TypeLiteral<TckRepository<StackImpl>>() {}).to(StackTckRepository.class);
+
+        bind(WorkspaceDao.class).to(JpaWorkspaceDao.class);
+        bind(StackDao.class).to(JpaStackDao.class);
+    }
+
+    public static class WorkspaceTckRepository extends JpaTckRepository<WorkspaceImpl> {
+        @Inject
```

Comment:
`JpaTckRepository` injects `EntityManager` and `UnitOfWork`, `JpaTckRepository` is an abstract class, so lets assume that `@Inject` is removed, how to bind `WorkspaceTckRepository` ? `bind(...).toInstance(new WorkspaceTckRepository())` won't work, as it won't inject fields to the superclass,  on the other hand i think it is not really good to use `requestInjection` for this purpose, so i prefer `@Inject` to `requestInjection`